src/main/java/clients/symphony/api/PresenceClient.java

```diff
@@ -40,35 +56,108 @@ public UserPresence getUserPresence(Long userId, boolean local) throws SymClient
         }
     }
 
-    public UserPresence setPresence(Presence status) throws SymClientException {
-        return setPresence(status.toString());
+    /**
+     * Returns the online status of the calling user.
+     *
+     * @return the calling user presence status
+     * @since 1.47
+     */
+    public UserPresence getUserPresence() throws SymClientException {
+        final Invocation.Builder builder = this.getTarget()
+            .path(PodConstants.GET_OR_SET_PRESENCE)
+            .request(MediaType.APPLICATION_JSON)
+            .header("sessionToken", botClient.getSymAuth().getSessionToken());
+
+        try (final Response response = builder.get()) {
+            if (response.getStatusInfo().getFamily() != Response.Status.Family.SUCCESSFUL) {
+                try {
+                    handleError(response, botClient);
+                } catch (UnauthorizedException ex) {
+                    return getUserPresence();
+                }
+                return null;
+            }
+            return response.readEntity(UserPresence.class);
+        }
     }
 
-    public UserPresence setPresence(String status) throws SymClientException {
-        Entity entity = Entity.entity(new Category(status), MediaType.APPLICATION_JSON);
+    /**
+     * Returns the presence of all users in a pod
+     * @param lastUserId Last user ID retrieved, used for paging. If provided, results skip users with IDs less than this parameter.
+     * @param limit Maximum number of records to return. The maximum supported value is 5000.
+     * @return a list of user presence
+     * @since 1.47
+     */
+    public List<UserPresence> getAllPresence(@Nullable Long lastUserId, @Nullable Integer limit) throws SymClientException {
+        WebTarget target = this.getTarget();
+
+        if (lastUserId != null) {
+            target = target.queryParam("lastUserId", lastUserId);
+        }
 
-        Invocation.Builder builder = botClient.getPodClient()
-            .target(botClient.getConfig().getPodUrl())
-            .path(PodConstants.SETPRESENCE)
+        if (limit != null) {
```

Comment:
if it is greater than 5000 why to invoke the api?

Reply:
Because the API doesn't enforce any limitation, so there's no reason to define it at SDK level

Reply:
Done : https://github.com/SymphonyPlatformSolutions/symphony-api-client-java/pull/89/commits/9e21f15ae87f1d96e6f1bc0e7a8a327cd840d152